oqamaint/openqabot.py

```diff
@@ -0,0 +1,588 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import print_function
+
+from collections import namedtuple
+from datetime import date
+import md5
+from pprint import pformat
+import re
+
+
+import requests
+import osc.core
+
+import ReviewBot
+from osclib.comments import CommentAPI
+
+from suse import SUSEUpdate
+
+try:
+    from xml.etree import cElementTree as ET
+except ImportError:
+    from xml.etree import ElementTree as ET
+
+
+try:
+    import simplejson as json
+except ImportError:
+    import json
+
+QA_UNKNOWN = 0
+QA_INPROGRESS = 1
+QA_FAILED = 2
+QA_PASSED = 3
+
+Package = namedtuple('Package', ('name', 'version', 'release'))
+pkgname_re = re.compile(r'(?P<name>.+)-(?P<version>[^-]+)-(?P<release>[^-]+)\.(?P<arch>[^.]+)\.rpm')
```

Comment:
Nearly identical to:

https://github.com/openSUSE/osc-plugin-factory/blob/4011159df44ce05b3c74a53e29ffe007d1b27a26/osclib/core.py#L15

`osclib/core.py` is intended for general `osc/core.py` related helpers. Stuff that is generalize to that level ought to live there. Many of the tools have duplicated lots and lots of methods which is why I started collecting them there (and adding new ones). This follows the same method signature as `osc/core.py` rather than `StagingAPI` requiring a lot more to initialize.

Reply:
nearly ... not same

Reply:
That's what refactoring is about, really need to copies of 99% identical regex where the prefix is different? The prefix `(?:.*::)?` being optional I would expect this to still work. This repo already has far too much copy/paste instead of refactoring and re-use...